parsing/parser.mly

```diff
@@ -2454,6 +2458,12 @@ let_binding_body:
       { let loc = ($startpos($1), $endpos($3)) in
         (ghpat ~loc (Ppat_constraint($1, $3)), $5) }
 ;
+let_binding_body:
+  | let_binding_body_no_punning
+      { let p,e = $1 in (p,e,false) }
+  | val_ident %prec below_HASH
+      { (mkpatvar ~loc:$loc $1, mkexpvar ~loc:$loc $1, true) }
```

Comment:
Nitpick: I suspect that if you had used a `punned_binding` rule for this case, you could have shared it with the `letop_binding_body` grammar. (The `%prec` annotation would stay here, I guess?)